src/core_modules/capture-core/src/components/QuickSelector/ProgramSelector.component.js

```diff
@@ -1,63 +1,209 @@
+    selectedPaper: {
+        backgroundColor: '#f6f6f6',
+        borderRadius: 5,
+        padding: 5,
+    },
+    selectedButton: {
+        float: 'right',
+        width: 20,
+        height: 20,
+    },
+    selectedButtonIcon: {
+        width: 20,
+        height: 20,
+    },
+    programAC: {
+        backgroundColor: 'white',
+        border: '1px solid lightGrey',
+        borderRadius: 5,
+        padding: '0px 5px 2px 5px',
     },
 });
 
 type Props = {
-    handleChangeProgram: (value: string) => void,
     handleClickProgram: (value: string) => void,
+    handleSetCatergoryCombo: (value: string, value: string) => void,
+    resetProgram: () => void,
     selectedProgram: Object,
+    selectedCategories: Object,
     classes: Object,
 };
 
 class ProgramSelector extends Component<Props> {
-    handleChange: (event: any) => void;
     handleClick: (program: Object) => void;
+    handleClickCategoryOption: (value: string, value: string) => void;
     constructor(props) {
         super(props);
-        this.handleChange = this.handleChange.bind(this);
         this.handleClick = this.handleClick.bind(this);
-    }
-
-    handleChange(event) {
-        this.props.handleChangeProgram(event.target.value);
+        this.handleClickCategoryOption = this.handleClickCategoryOption.bind(this);
     }
 
     handleClick(program) {
         this.props.handleClickProgram(program.id);
     }
 
-    // TODO: Add support for cat-combos.
+    handleClickCategoryOption(selectedCategoryOption, categoryId) {
+        this.props.handleSetCatergoryCombo(selectedCategoryOption, categoryId);
+    }
+
+    handleResetProgram() {
+        this.props.resetProgram();
+        
+        // If categoryOptions have been selected, they need to be removed when program is de-selected.
+        if (this.props.selectedCategories) {
+            this.props.handleSetCatergoryCombo(null, null);
+        }
+    }
+
+    handleResetCategory(categoryId) {
+        this.props.handleSetCatergoryCombo(null, categoryId);
+    }
+
     render() {
         const programsArray = Array.from(programs.values());
+        
+        const programOptions = programsArray.map(optionCount => new Option((_this) => {
+            _this.value = optionCount.id;
+            _this.text = optionCount.name;
+        }));
+
+        const programOptionSet = new OptionSet('programOptionSet', programOptions);
+
+        // If program is set in Redux state.
+        if (this.props.selectedProgram) {
+            let selectedProgram = {};
+            for (let i = 0; i < programsArray.length; i++) {
+                // Get full program object based on id from this.props.selectedProgram.
+                if (programsArray[i].id === this.props.selectedProgram) {
+                    selectedProgram = programsArray[i];
+                }
+            }
+            if(selectedProgram.categoryCombo && selectedProgram.categoryCombo.categories.length > 0 && !selectedProgram.categoryCombo.isDefault) {
```

Comment:
Not necessarily something we should do right now - but the category options might have been split out into another component or file.